### PR TITLE
BasetTxnID type extraction

### PR DESCRIPTION
**Pull request by awskii**

### CODE REVIEW

File: core/types/block.go [hunk 1]
```diff
@@ -640,8 +640,43 @@ type RawBody struct {
 	Requests     Requests
 }
 
+// Represents original transaction number in blockchain (do not includes system txs in block)
+type BaseTxID uint64
+
+func CapBaseTxID(tx int) uint64 {
```
Comment (AskAlexSharov) [inline core/types/block.go:646]: `CapBaseTxID` - don't understand func name (can't translate).

File: core/types/block.go [hunk 2]
```diff
@@ -640,8 +640,43 @@ type RawBody struct {
 	Requests     Requests
 }
 
+// Represents original transaction number in blockchain (do not includes system txs in block)
```
Comment (AskAlexSharov) [inline core/types/block.go:643]: 1. ` Represents original transaction number in blockchain` - no. by naming (see `Dev.md`): 
- Num - it's blockchain-related number - same on all nodes
- ID - it's our internal auto-increment - not related to blockchain and different on all nodes. 

2. `BaseTxID` let's rename it to `BaseTxnID`. Also let's rename `txID` to `txnID`. (also by naming - see `Dev.md`)

File: core/types/block.go [hunk 3]
```diff
@@ -640,8 +640,43 @@ type RawBody struct {
...
+// From the other hand, TxAmount includes 2 system txs as well as all txs in block.
+func (b BaseTxID) Last(txns int) uint64 { return b.First() + uint64(txns) }
+
+// FirstSystemTx returns first system tx number in block
+func (b BaseTxID) FirstSystemTx() BaseTxID { return b }
+
+// LastSystemTx returns last system tx number in block. result+1 will be baseID of next block a.k.a. beginning system tx number
+// eg txAmount = 3+2/*systemTx*/ = 5 therefore:
+//
+//	0 - base tx/systemBegin
```
Comment (AskAlexSharov) [inline core/types/block.go:671]: sustem transactions are used to write history of state changes done by consensus (not by eth-transactions) - for example "miner rewards"

**Review by AskAlexSharov — CHANGES_REQUESTED** [review]

**Review by AskAlexSharov — APPROVED** [review]

**Comment by awskii** [comment]
amoy file matches. waiting for building files for sepolia to be sure.

**Comment by awskii** [comment]
I triple-checked with amoy and double with sepolia, headers/bodies/txns .seg come out all the same, state files also matched for sepolia

**Pull request merged**